### ISSUE TITLE
PSR2.Methods.FunctionCallSignature false positive when arrow function has array return type

### ISSUE THREAD

**Issue by jasonmm** [squizlabs/PHP_CodeSniffer#2773]
This might be related to #2523.

Using the file:
```php
<?php

declare(strict_types=1);

print_r(array_map(
    fn(string $n): array => [1, bcadd($n, '3')],
    ['1', '2', '3']
));
```

Produces the error:

```bash
$ phpcs --standard=PSR12 -s ./test2.php

FILE: test2.php
-------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------
 6 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
   |       |     (PSR2.Methods.FunctionCallSignature.MultipleArguments)
-------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------------------

Time: 54ms; Memory: 6MB

```

This seems specific to an arrow function that has an `array` return type with a multiple argument function call in the body of the arrow function. Using `fn(string $n): string => bcadd($n, '3')` or `fn(string $n): array => [$n, 3]` as the arrow function produces no error from phpcs.

**Comment by gsherwood** [comment]
Thanks for the report. The fix will be in 3.5.4.

**Comment by cl1ck** [comment]
When an arrow function with an array return value is the last function argument, this still reports an incorrect error. I have the following in my code:
```PHP
return $this->mapResults(
    $dql,
    fn (array $row) => [$row[0], $row[3]]
 );
```

**Comment by jrfnl** [comment]
@cl1ck You may want to open a new issue as 1) this is closed and 2) your issue is actually a different one.

Your issue can be reproduced even without the `array` type declaration and without the first argument.
As far as I can see, it has to do with the `,` within the short array confusing the sniff into thinking there are multiple arguments.

**Comment by gsherwood** [comment]
I opened issue #2895 for the issue above. Someone else also commented on another closed issue with the same general problem.
